server/webapp/WEB-INF/rails.new/app/presenters/api_v1/stage_representer.rb

```diff
@@ -0,0 +1,94 @@
+##########################################################################
+# Copyright 2015 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV1
+  class StageRepresenter < ApiV1::BaseRepresenter
+    alias_method :stage, :represented
+
+    link :self do |opts|
+      if stage.getIdentifier
+        opts[:url_builder].apiv1_stage_instance_by_counter_api_url(pipeline_name:    stage.getIdentifier.getPipelineName,
+                                                                   pipeline_counter: stage.getIdentifier.getPipelineCounter,
+                                                                   stage_name:       stage.getName,
+                                                                   stage_counter:    stage.getCounter)
+      end
+    end
+
+    link :doc do |opts|
+      'http://api.go.cd/#get-stage-instance'
+    end
+
+    property :getName, as: :name
+    property :result, exec_context: :decorator
+    property :getCounter, as: :counter
+    property :getApprovalType, as: :stage_type
+    property :rerun_of_stage, exec_context: :decorator, decorator: StageSummaryRepresenter
```

Comment:
@arvindsv @jyotisingh 
Can you provide context for how this value is set.

In the the old API
* When the stage is re-run this value is null
* When a job is re-run the value points to first counter of the stage

It's the same in the new API. Is that correct?